InteractionBase/src/handlers/GenericHandler.cpp

```diff
@@ -424,6 +425,8 @@ void GenericHandler::keyPressEvent(Visualization::Item *target, QKeyEvent *event
 			// Check if the parent of the node is a list and if so, delete this node
 			if (auto list = DCast<Model::List>(node->parent()))
 			{
+				for (auto view : target->scene()->viewItems()->viewItemsAsList())
```

Comment:
There is `Scene::nodesUpdated` which gets called whenever some nodes have been updated. You should try to work with that method. It receives a list of updated nodes. I'm not sure if this list will contain deleted nodes, or only the parents of deleted nodes, so you'd have to check. In any case, don't do too much work directly in this method. Just call something inside `ViewItemManager` that will do most of the work.